xmtp_proto/src/api_client.rs

```diff
@@ -86,11 +89,19 @@ pub trait XmtpApiSubscription {
     fn close_stream(&mut self);
 }
 
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+pub trait MutableApiSubscription {
+    async fn update(&mut self, req: SubscribeRequest) -> Result<(), Error>;
+    fn close(&self);
+}
+
 // Wasm futures don't have `Send` or `Sync` bounds.
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait XmtpApiClient {
     type Subscription: XmtpApiSubscription;
+    type MutableSubscription: MutableApiSubscription + Stream<Item = Result<Envelope, Error>>;
```

Comment:
Might be cleaner/consistent to do:
```
pub trait MutableApiSubscription: Stream<Item = Result<Envelope, Error>> {
```

on L94 above, and

```
type MutableSubscription: MutableApiSubscription;
```

here. Then all implementers of `MutableApiSubscription` automatically must implement `Stream`

Reply:
That's a great idea. Updated.